stdlib/public/core/CollectionAlgorithms.swift

```diff
@@ -241,6 +242,11 @@ extension MutableCollection {
   ///     let second = numbers[p...]
   ///     // second == [60, 40]
   ///
+  /// Note that elements in both partitions have changed their
+  /// relative order compared to before partitioning. That is, `40`
+  /// appears before `60` in the original collection, but afterward
+  /// after calling `partition(by:)`.
```

Comment:
How about this, to avoid "afterward after" which could be hard to read, and to avoid "relative order" which seems redundant:

```suggestion
  /// Note that the order of elements elements in both partitions changed.
  /// That is, `40` appears before `60` in the original collection,
  /// but, after calling `partition(by:)`, `60` appears before `40`.
```